libraries/lib-numeric-formats/ProjectTimeSignature.cpp

```diff
@@ -0,0 +1,123 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/**********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file ProjectTimeSignature.cpp
+
+ Dmitry Vedenko
+
+ **********************************************************************/
+#include "ProjectTimeSignature.h"
+
+#include "Beats.h"
+#include "Project.h"
+#include "XMLAttributeValueView.h"
+#include "XMLWriter.h"
+
+static const AttachedProjectObjects::RegisteredFactory key
+{
+   [](AudacityProject &project)
+   { return std::make_shared<ProjectTimeSignature>(); }
+};
+
+ProjectTimeSignature& ProjectTimeSignature::Get(AudacityProject& project)
+{
+   return project.AttachedObjects::Get<ProjectTimeSignature&>(key);
+}
+
+const ProjectTimeSignature&
+ProjectTimeSignature::Get(const AudacityProject& project)
+{
+   return Get(const_cast<AudacityProject&>(project));
+}
```

Comment:
I see you have got used to the boilerplate for making new attached objects.

`ClientData::Site` is a great idea if I may say so immodestly, but I find this kind of repetition a bit painful.

I have some ideas for using a CRTP template to automate the writing of all this `Get` junk, requiring only that you supply the factory.

Please review my #4395 .  That, or something close to it, may be part of the solution.